Assets/Scripts/Score/SessionScoreHandler.cs

```diff
@@ -0,0 +1,55 @@
+using UnityEngine.Events;
+using UnityEngine;
+using TMPro;
+
+public class SessionScoreHandler : MonoBehaviour , ScoreSystem
+{
+    private static SessionScoreHandler _scoreHandler;
+
+    [SerializeField] private TextMeshProUGUI _scoreText;
+    [SerializeField] private ObstacleValue _obstacleValues;
+
+
+    private int _score;
+    public float _scoreMultiplier = 1;
+    private void Awake()
+    {
+        _scoreHandler = this;
+    }
+    public void AddPoints(string value)
+    {
+        //will parse the string into a value
+        ObstacleTypes type = value.Parse<ObstacleTypes>();
+        switch (type)
+        {
+            case ObstacleTypes.Coin:
+                _score += (int)(_obstacleValues.Coins * _scoreMultiplier);    
```

Comment:
Misschien ipv hier point calculation te doen een aparte method aanmaken. dan hoeven we niet bij elke case * multiplier te doen. wat je misschien kan doen is met scriptable object de enum value direct zetten. dan hoev je aleen in de nieuwe method te doen: type * _scoreMultiplier